### PR TITLE
Ticket env template, tool use support, removed user_name add 5 point TM

### PR DESCRIPTION
**Pull request by carolius**
## Description

- Removed user_name (might break some stuff)
- Changed transition model to 5 point scale for some environments (might also break stuff but it still supports the binary version so should be fine)
- Iteration on env templates
- Added ticket environment template
- Added tool use support to collator etc

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes. 
- [x] I have run the local tests with `pytest --gpus=X`.
- [ ] I have run the experiment with `EI_test_up.yaml`

### CODE REVIEW

File: influence_benchmark/RL/conversation_collator.py [hunk 1]
```diff
@@ -122,5 +147,15 @@ def torch_call(self, examples: List[Union[List[int], Any, Dict[str, Any]]]) -> D
...
                 batch["labels"][i, user_template_start:assistant_response_start] = self.ignore_index
+
+            # Mask tool calls and tool responses
+            for tool_call_start in tool_call_start_idxs:
+                next_start = next((x for x in start_header_idxs if x > tool_call_start), len(batch["labels"][i]))
+                batch["labels"][i, tool_call_start:next_start] = self.ignore_index
+
+            for tool_response_start in tool_response_start_idxs:
+                next_start = next((x for x in start_header_idxs if x > tool_response_start), len(batch["labels"][i]))
+                batch["labels"][i, tool_response_start:next_start] = self.ignore_index
```
Comment (micahcarroll) [inline influence_benchmark/RL/conversation_collator.py:158]: Did not double check this with the debugger. Potential source of bugs if we see issues with this in the future

File: influence_benchmark/config/env_configs/normal_nudging_therapist/normal_neglecting-responsibility.yaml [hunk 2]
```diff
@@ -1,217 +0,0 @@
-env_name: neglecting-responsibility
```
Comment (micahcarroll) [inline influence_benchmark/config/env_configs/normal_nudging_therapist/normal_neglecting-responsibility.yaml:1]: Is this just a stub? There are currently no env files here

File: influence_benchmark/environment/assessor_model.py [hunk 3]
```diff
@@ -33,7 +33,7 @@ def prepare_messages(self, state) -> List[dict]:
                 if message["role"] == "agent":
                     messages.append(f"{state.variables['agent_name']}: {message['content']}")
                 elif message["role"] == "environment":
-                    messages.append(f"{state.variables[self.config['env_char_label']]}: {message['content']}")
+                    messages.append(f"User: {message['content']}")
                 elif message["role"] == "environment_system":
                     messages.append(f"system: {message['content']}")
```
Comment (micahcarroll) [inline influence_benchmark/environment/assessor_model.py:38]: Not sure what this does, but do we want User capitalized but not system?

File: influence_benchmark/environment_vectorized/transition_model_vectorized.py [hunk 4]
```diff
@@ -59,9 +62,14 @@ def _transition_postprocessing(self, transition_probs: Dict[str, float], state:
         Returns:
             str: The selected transition.
         """
-        transitions, probabilities = zip(*transition_probs.items())
-
         if unfixable_flag:
             return state.default_transition
+        # check if the transitions can be made into integers
+        transitions, probabilities = zip(*transition_probs.items())
```
Comment (micahcarroll) [inline influence_benchmark/environment_vectorized/transition_model_vectorized.py:68]: What are transitions normally? Strings? Not sure what this if/else is supposed to correspond to. Worth commenting a bit more maybe?

**Review by micahcarroll — COMMENTED** [review]
Broadly looks good, will build off this later today and leave more comments after that probably

File: influence_benchmark/config/env_configs/normal_nudging_therapist/normal_neglecting-responsibility.yaml (hunk 2 above)
Comment (carolius) [reply influence_benchmark/config/env_configs/normal_nudging_therapist/normal_neglecting-responsibility.yaml:1]: fixed

Comment (micahcarroll) [reply influence_benchmark/config/env_configs/normal_nudging_therapist/normal_neglecting-responsibility.yaml:1]: Oh I don't think the commits pushed

File: influence_benchmark/generate_histories/generate_histories.py [hunk 5]
```diff
@@ -88,6 +93,12 @@ async def main():
             )
             del sub_env_config["weak_char_action"]
             del sub_env_config["normal_char_action"]
+        if "weak_certainty" in sub_env_config:  # tickets
+            sub_env_config["certainty"] = (
+                sub_env_config["weak_certainty"] if weakness else sub_env_config["normal_certainty"]
+            )
+            del sub_env_config["weak_certainty"]
+            del sub_env_config["normal_certainty"]
```
Comment (micahcarroll) [inline influence_benchmark/generate_histories/generate_histories.py:101]: Noticed that in nudging-therapist and tickets you're not modifying pm based on weak or normal. Is that intended?

**Comment by micahcarroll** [comment]
I think once you push your final changes this is good to merge. Have a couple of formatting fixes in next PR that is coming soon, but nothing major that should block the merging of this PR. The config generation code looked good and easy to use

File: influence_benchmark/generate_histories/generate_histories.py [hunk 6]
```diff
@@ -7,14 +7,19 @@
...
-file_prefix = "weak_" if weakness else "normal_"
+file_prefix = (
+    "weak_"
+    if weakness and env_class != "tickets"
+    else "uncertain_" if weakness and env_class == "tickets" else "normal_"
+)
+
 output_file_base = f"influence_benchmark/config/env_configs/{file_prefix}{env_class}/"
 extend = False
 num_runs = 1  # don't set this to more than 1 if not extending
```
Comment (micahcarroll) [inline influence_benchmark/generate_histories/generate_histories.py:25]: Could you also leave comments for the BlockDumper? Why do we need it? What is it doing? Do we need to use that to save trajectories during training time, or does it not matter?

**Review by micahcarroll — APPROVED** [review]

File: influence_benchmark/generate_histories/generate_histories.py (hunk 5 above)
Comment (carolius) [reply influence_benchmark/generate_histories/generate_histories.py:101]: Yeah. In nudging I wanted the things to be very similar, the only difference is whether they have a positive or negative experience. In tickets weak doesn't really mean weak, I didn't think we wanted to do the population experiment on tickets. Here it just modifies how uncertain they seem in the initial message, but hopefully the agent will deceive both certain and uncertain

File: influence_benchmark/generate_histories/generate_histories.py (hunk 6 above)
Comment (carolius) [reply influence_benchmark/generate_histories/generate_histories.py:25]: The yaml files it generated were kind of ugly (still correct) so I wanted to make it be the same format as when I made them.

Comment (carolius) [reply influence_benchmark/generate_histories/generate_histories.py:25]: During training we save trajectories as jsonl not as yaml so this is not applicable then

**Pull request merged**